fee_allocator/accounting/distribution.py

```diff
@@ -112,66 +101,36 @@ def re_distribute_incentives(
             # Calculate pool weight:
             pool_weight = _pool_weights[pool_id_to_receive]
             # Calculate incentives to receive
-            to_receive = round(incentives_to_redistribute * pool_weight, 2)
-            to_receive_aura = round(incentives_to_redistribute_aura * pool_weight, 2)
-            to_receive_bal = round(incentives_to_redistribute_bal * pool_weight, 2)
-            # Need to check if aura or bal incentives
-            # are less than min aura incentive and distribute accordingly
-            if _data_to_receive['aura_incentives'] + to_receive_aura < min_aura_incentive:
-                incentives[pool_id_to_receive]['bal_incentives'] += to_receive
-            elif _data_to_receive['bal_incentives'] + to_receive_bal < min_aura_incentive:
-                incentives[pool_id_to_receive]['aura_incentives'] += to_receive
-            else:
-                # In case both are > min aura incentive, we distribute evenly
-                incentives[pool_id_to_receive]['aura_incentives'] += to_receive_aura
-                incentives[pool_id_to_receive]['bal_incentives'] += to_receive_bal
+            to_receive = round(incentives_to_redistribute * pool_weight, 4)
+            to_receive_aura = round(incentives_to_redistribute_aura * pool_weight, 4)
+            to_receive_bal = round(incentives_to_redistribute_bal * pool_weight, 4)
+            incentives[pool_id_to_receive]['aura_incentives'] += to_receive_aura
+            incentives[pool_id_to_receive]['bal_incentives'] += to_receive_bal
             incentives[pool_id_to_receive]['total_incentives'] += to_receive
             incentives[pool_id_to_receive]['redirected_incentives'] += to_receive
-
-    # Now, go through each pool once again and redistribute aura and bal incentives considering aura_vebal_share
+    ## Now after everything is done, we need to make sure that all pools have at least min_aura_incentive
```

Comment:
At the very end we consider the aura_min.  First we just move everything over to BAL to respect the min, keeping track of how much we move.